packages/rspack/src/Module.ts

```diff
@@ -48,26 +49,33 @@ export class Module {
 	#inner: JsModule;
 	_originalSource?: Source;
 
+	context?: string;
+	resource?: string;
+	request?: string;
+	userRequest?: string;
 	rawRequest?: string;
 
-	static __from_binding(module: JsModule) {
-		return new Module(module);
+	buildInfo: Record<string, any>;
+	buildMeta: Record<string, any>;
+
+	static __from_binding(module: JsModule, compilation?: Compilation) {
+		return new Module(module, compilation);
 	}
 
-	constructor(module: JsModule) {
+	constructor(module: JsModule, compilation?: Compilation) {
 		this.#inner = module;
+		this.context = module.context;
+		this.resource = module.resource;
+		this.request = module.request;
+		this.userRequest = module.userRequest;
 		this.rawRequest = module.rawRequest;
-	}
-
-	get context(): string | undefined {
-		return this.#inner.context;
-	}
 
-	get resource(): string | undefined {
-		return this.#inner.resource;
+		const customModule = compilation?.getCustomModule(module.moduleIdentifier);
+		this.buildInfo = customModule?.buildInfo || {};
+		this.buildMeta = customModule?.buildMeta || {};
 	}
 
-	get originalSource(): Source | null {
+	originalSource(): Source | null {
```

Comment:
seems to be a breaking change to plugin developer, cc @h-a-n-a

Reply:
I think rspack makes the mistake. See https://github.com/webpack/webpack/blob/ac8e83a4a2984ed977fd6c23a609da5cb43da492/lib/NormalModule.js#L1388